### PR TITLE
cleanup(deps): RHICOMPL-3853 replace ruby-kafka with rdkafka

### PR DESCRIPTION
**Pull request by RoamingNoMaD**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
@vkrizan can you please also take a look?

File: app/producers/application_producer.rb [hunk 1]
```diff
@@ -30,42 +38,29 @@ def logger
...
+      return {} unless Settings.kafka.security_protocol.downcase == 'sasl_ssl'
 
-      config = {
-        sasl_prefix(:username) => Settings.kafka.sasl_username,
-        sasl_prefix(:password) => Settings.kafka.sasl_password,
-        :ssl_ca_certs_from_system => true
+      {
+        'sasl.username' => Settings.kafka.sasl_username,
+        'sasl.password' => Settings.kafka.sasl_password,
+        'sasl.mechanism' => Settings.kafka.sasl_mechanism
```
Comment (vkrizan) [inline app/producers/application_producer.rb:50]: I think we would need to set `security.protocol` as well.
Possible values are `plaintext, ssl, sasl_plaintext, sasl_ssl`, as per (CONFIGURATION.md)[https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md].

File: app/producers/application_producer.rb [hunk 2]
```diff
@@ -20,7 +28,7 @@ def deliver_message(msg)
         service: SERVICE,
         source: ENV['APPLICATION_TYPE']
       )
-      kafka&.deliver_message(msg.to_json, topic: self::TOPIC)
+      kafka&.produce(payload: msg.to_json, topic: self::TOPIC)&.wait
```
Comment (vkrizan) [inline app/producers/application_producer.rb:31]: JFYI: Minimum wait time is 0.1s and max 60s.
ref: https://www.rubydoc.info/github/appsignal/rdkafka-ruby/Rdkafka/AbstractHandle#wait-instance_method

File: app/producers/application_producer.rb [hunk 3]
```diff
@@ -12,6 +14,12 @@ class ApplicationProducer < Kafka::Client
   #   TOPIC = 'platform.payload-status'
 
   class << self
+    def ping
+      # The partition count method fails if the connection is not alive, so we are
+      # sending a random topic name to it for status checks.
+      kafka.partition_count(Settings.kafka_producer_topics.to_h.values.compact.sample)
```
Comment (vkrizan) [inline app/producers/application_producer.rb:20]: Where do we get the `Settings.kafka_producer_topics` from?
Do we need to rehash it?

Comment (RoamingNoMaD) [reply app/producers/application_producer.rb]: `Settings.kafka.topics` is indeed convertible to a Hash. See my latest [push](https://github.com/RedHatInsights/compliance-backend/pull/1752/files#diff-8ccd2dde6a3d162300523af2c9d5fb197f1c707042fcc63072e20b750af10f5aR20).

**Pull request merged**